bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/FileSystemUpgrade.java

```diff
@@ -99,7 +99,12 @@ public boolean accept(File dir, String name) {
     private static List<File> getAllDirectories(ServerConfiguration conf) {
         List<File> dirs = new ArrayList<>();
         dirs.addAll(Lists.newArrayList(conf.getJournalDirs()));
-        Collections.addAll(dirs, conf.getLedgerDirs());
+        final File[] ledgerDirs = conf.getLedgerDirs();
+        final File[] indexDirs = conf.getIndexDirs();
+        if (indexDirs != null && indexDirs != ledgerDirs) {
```

Comment:
We can't use `!=` to check if two array equals or not

Reply:
@hangc0276 I have addressed your comments. Please take a look. Thanks.